source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+  the error contains all of the specified labels (e.g. using the
+  ``hasErrorLabel`` method).
+
+- ``errorLabelsOmit``: Optional array of strings. A list of error label strings
+  that the error is expected not to have. The test runner MUST assert that
+  the error does not contain any of the specified labels (e.g. using the
+  ``hasErrorLabel`` method).
+
+.. _expectedError_expectedResult:
+
+- ``expectedResult``: Optional mixed type. This field follows the same rules as
+  `operation.expectedResult <operation_expectedResult_>`_ and is only used in
+  cases where the error includes a result (e.g. `bulkWrite`_). If specified, the
+  test runner MUST assert that the error includes a result and that it matches
+  this value.
+
+
+expectedEvent
+~~~~~~~~~~~~~
+
+An event (e.g. APM), which is expected to be observed while executing the test's
+operations.
+
+This document MUST contain **exactly one** top-level key that identifies the
+event type and maps to a nested document, which contains one or more assertions
+for the event's properties.
+
+The structure of this document is as follows:
+
+.. _expectedEvent_commandStartedEvent:
+
+- ``commandStartedEvent``: Optional document. Assertions for a one or more
+  `CommandStartedEvent <../command-monitoring/command-monitoring.rst#api>`__
+  fields.
+
+  The structure of this document is as follows:
+
+  - ``command``: Optional document. Test runners MUST follow the rules in
+    `Evaluating Matches`_ when processing this assertion.
+
+  - ``commandName``: Optional string. Test runners MUST assert that the actual
+    command name matches this value using a case-insensitive comparison.
+
+  - ``databaseName``: Optional string. Test runners MUST assert that the actual
+    command name matches this value using a case-insensitive comparison. THe
+    YAML file SHOULD use an `alias node`_ for this value (e.g.
+    ``databaseName: *database0Name``).
+
+.. _expectedEvent_commandSucceededEvent:
+
+- ``commandSucceededEvent``: Optional document. Assertions for a one or more
+  `CommandSucceededEvent <../command-monitoring/command-monitoring.rst#api>`__
+  fields.
+
+  The structure of this document is as follows:
+
+  - ``reply``: Optional document. Test runners MUST follow the rules in
+    `Evaluating Matches`_ when processing this assertion.
+
+  - ``commandName``: Optional string. Test runners MUST assert that the actual
+    command name matches this value using a case-insensitive comparison.
+
+.. _expectedEvent_commandFailedEvent:
+
+- ``commandFailedEvent``: Optional document. Assertions for a one or more
+  `CommandFailedEvent <../command-monitoring/command-monitoring.rst#api>`__
+  fields.
+
+  The structure of this document is as follows:
+
+  - ``commandName``: Optional string. Test runners MUST assert that the actual
+    command name matches this value using a case-insensitive comparison.
+
+
+collectionOrDatabaseOptions
```

Comment:
I am privy to calling this `EntityOptions` and opening this up to allow client to take these as well. Would we want to include `session` as well?

Reply:
In my earlier comment, I explained that this structure is defined purely to reduce duplication and doesn't correspond to a structure in any driver API. Can this be resolved in light of that?

No entities take a session, so I don't think there'd be any reason to add that here.

Reply:
@reggi ping****